### ISSUE TITLE
pytz still using IANA  Time Zone Database 2021c

### ISSUE THREAD

**Issue by FObersteiner** [stub42/pytz#70]
the current version Time Zone Database is [2021e](https://www.iana.org/time-zones), so 2021c seems outdated. This might cause invalid UTC offsets like e.g.

```python
from datetime import datetime
import pytz

>>> dt = datetime.fromisoformat('2046-05-31 22:00:00+00:00').astimezone(pytz.timezone("Europe/Paris"))
>>> print(dt)
2046-05-31 23:00:00+01:00
# should be 
# 2046-06-01 00:00:00+02:00

>>> print(pytz.__version__)
2021.3
```

```python
from zoneinfo import ZoneInfo
>>> dt = datetime.fromisoformat('2046-05-31 22:00:00+00:00').astimezone(ZoneInfo("Europe/Paris"))
>>> print(dt)
2046-06-01 00:00:00+02:00

>>> import tzdata
>>> print(tzdata.IANA_VERSION)
2021e
```

*Python 3.9.4 on Windows 10*

**Comment by FObersteiner** [comment]
related ? https://github.com/stub42/pytz/issues/31

**Comment by stub42** [comment]
Released 2022.1, including 2022a timezone database. No code changes, just the new db.

**Comment by FObersteiner** [comment]
can be closed; pytz now using most recent IANA db. Year 2038 problem remains though, see related issues.
